server/src/main/java/org/opensearch/tracing/Tracer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import java.io.Closeable;
+
+/**
+ * Tracer is the interface used to create a {@link Span} and interact with current active {@link Span}.
+ * It automatically handles the context propagation between threads, tasks, nodes etc.
+ *
+ * All methods on the Tracer object are multi-thread safe.
+ */
+public interface Tracer extends Closeable {
+
+    /**
+     * Starts the {@link Span} with given name and level
+     *
+     * @param spanName span name
+     * @param level span tracing level
+     */
+    void startSpan(String spanName, Level level);
+
+    /**
+     * Ends the current active {@link Span}
+     *
+     */
+    void endSpan();
+
+    /**
+     * Returns the current active {@link Span}
+     *
+     * @return current active span
+     */
+    Span getCurrentSpan();
```

Comment:
why do we need to return the current span otherwise it would defeat the your purpose of operating on span via TraceManager

Reply:
We had added this because we needed it for certain cases around creating spans for tasks. Will remove it.

Reply:
Done